pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py

```diff
@@ -130,6 +138,53 @@ def configure_logger(self, logger):
             else:
                 self.trainer.logger = logger
 
+    def cache_training_step_metrics(self, opt_closure_result):
+        """
+        This function is responsible to update
+        logger_connector internals metrics holder based for depreceated logging
+        """
+        using_results_obj = isinstance(opt_closure_result.training_step_output, Result)
+
+        # temporary dict to collect metrics
+        logged_metrics_tmp = {}
+        pbar_metrics_tmp = {}
+        callback_metrics_tmp = {}
+
+        if using_results_obj:
+            batch_log_metrics = opt_closure_result.training_step_output.get_batch_log_metrics(
+                include_forked_originals=False
+            )
+            logged_metrics_tmp.update(batch_log_metrics)
+
+            batch_pbar_metrics = opt_closure_result.training_step_output.get_batch_pbar_metrics(
+                include_forked_originals=False
+            )
+            pbar_metrics_tmp.update(batch_pbar_metrics)
+
+            forked_metrics = opt_closure_result.training_step_output.get_forked_metrics()
+            callback_metrics_tmp.update(forked_metrics)
+            callback_metrics_tmp.update(logged_metrics_tmp)
+
+        else:
+            batch_log_metrics = opt_closure_result.training_step_output.log_metrics
+            logged_metrics_tmp.update(batch_log_metrics)
+
+            callback_metrics = opt_closure_result.training_step_output.callback_metrics
+            callback_metrics_tmp.update(callback_metrics)
+
+            batch_pbar_metrics = opt_closure_result.training_step_output.pbar_on_batch_end
+            pbar_metrics_tmp.update(batch_pbar_metrics)
+
+        # track progress bar metrics
+        if len(pbar_metrics_tmp) > 0:
+            self.add_progress_bar_metrics(pbar_metrics_tmp)
+
+        self.callback_metrics.update(callback_metrics_tmp)
+
+        # save legacy log metrics
+        self.logged_metrics.update(logged_metrics_tmp)
+        self.cached_results("train").legacy_batch_log_metrics.update(logged_metrics_tmp)
```

Comment:
I see this key being used everyone and was just curious on the design choice here. Would it be better to have a singular 'train' key defined somewhere and use this to refer to the cached_results? The reason being right now we have "train" written in the code in multiple places, and hardcoding strings is typically not a good thing!

That being said, I think I can be convinced that storing a key like `self.cached_result_key = "train"` isn't a worthwhile endeavour.

Reply:
I could use the `self._current_stage` to do so. You are right

Reply:
Done !